acceptance/tests/language/sensitive_data_type.rb

```diff
@@ -75,13 +82,20 @@
           run_assertions(assertion_code, result)
         end
 
-        if agent['platform'] =~ /fedora|centos|el|redhat|scientific|ubuntu|debian|cumulus/
-          step "assert no redacted data in syslog" do
-            #sigh.  gee, thanks for the help, beaker!
-            result = dump_puppet_log(agent)
-            raise if result.length < 3
-            string_io_index = 2
-            run_assertions(assertion_code, result[string_io_index].string)
+        step "assert no redacted data in syslog" do
+          #sigh.  gee, thanks for the help, beaker!
+          syslogfile = case agent['platform']
+                         when /fedora|centos|el|redhat|scientific/ then
+                           '/var/log/messages'
+                         when /ubuntu|debian|cumulus/ then
+                           '/var/log/syslog'
+                         else
+                           nil
+                       end
+          if syslogfile
+            result = agent.exec(Command.new("tail -n 100 #{syslogfile}"),
+                                :acceptable_exit_codes => [0, 1]).stdout.chomp
+            run_assertions(refutation_code, result)
```

Comment:
Maybe it would also make sense to rework this test to see that if `puppet agent` is run with the `--logdest` set to `syslog` instead of defaulting to `console` that both the refutations of sensitive content pass and that expected debug output that is not sensitive is present in the syslog?  Seems like at least ensuring that the syslog doesn't have sensitive content in it may be enough, though, since both the assertions / refutations are run against the puppet agent console output as well.